climpred/stats.py

```diff
@@ -168,14 +172,15 @@ def rm_poly(ds, order, dim='time'):
     # check if there's any NaNs in the provided dim because
     # interpolate_na is computationally expensive to run regardless of NaNs
     if any(nan_locs.sum(axis=0)) > 0:
+        # Could do a check to see if there's any NaNs that aren't bookended.
```

Comment:
Note that before this was wrapped at the end of an if -> elif -> else chain. So basically only backfill was being called everywhere.